indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -474,31 +467,180 @@ sealed class Asn1Structure(
      */
     val isActuallySorted: Boolean by if (sortChildren) lazyOf(true) else lazy { children.sortedBy { it.tag } == children }
 
-    private var index = 0
+    private val iterator by lazy { iterator() }
 
     /**
      * Returns the next child held by this structure. Useful for iterating over its children when parsing complex structures.
      * @throws [Asn1StructuralException] if no more children are available
+     *### Migration Examples:
+     *
+     * Using `decodeRethrowing`:
+     * ```kotlin
+     * val result = structure.decodeRethrowing {
```

Comment:
do we expect consumers to use `decodeRethrowing`?

`runRethrowing`/`decodeRethrowing` are internal-use things that convert exceptions, right?

Reply:
decodeRethrowing is handy externally as well